src/app/components/common/node-editor/editor.ts

```diff
@@ -335,25 +335,29 @@ export function getUpdatedFilterPipeline() {
 
   for (let i = 0; i < nodes.length; i++) {
     if (i == 0) {
+      // check if starting node of pipeline i.e. south/storage is connected
       if (!connections.find(c => c.source === nodes[i].id)) {
         console.log("Dangling connection");
         return false;
       }
     }
     else if (i == 1) {
+      // check if last node of pipeline i.e. storage/north is connected
       if (!connections.find(c => c.target === nodes[i].id)) {
         console.log("Dangling connection");
         return false;
       }
     }
     else {
+      // check if all the filters are connected from both sides
       if (!connections.find(c => c.source === nodes[i].id) || !connections.find(c => c.target === nodes[i].id)) {
         console.log("Dangling connection");
         return false;
       }
     }
   }
 
+  // check for self loop
```

Comment:
self loop?

Reply:
This check is to test if there is a connection from input socket to output socket of same node.